### ISSUE TITLE
Put support for labels into RRDHOST

### ISSUE THREAD

**Issue by amoss** [netdata/netdata#7408]
* Insert a linked list into the RRDHOST structure that holds key/value pairs.
* Insert locks to protect access to the list.
* Perform very basic tests that we can add key/value pairs and iterate over them.

This functionality is not accessible from anywhere yet, CMocka is a good way to do this testing.

**Comment by thiagoftsm** [comment]
I do not think it is only key/value, I am in the group of people that also believe that our users can insert thousands of labels, so it will be necessary to insert an `avl` element on top of the structure that will be used to organize labels and tags, in other words, I think the structure needs to be like `rrdcalc` and `rrdset`.

**Comment by ilyam8** [comment]
> our users can insert thousands of labels

What makes you think so? Any real use case?

**Comment by thiagoftsm** [comment]
Ok, I agree I was very pessimist, you are correct, but let us bring a more realistic situation, case our users have 32 labels and we wanna get the label that is in the position 31. Case we use only link list we will need to advance 30 times comparing values to get the data, on the other hand, case we have an `avl` we will need 5 comparisons.

**Comment by amoss** [comment]
This was resolved in the technical refinement meeting. There is no use-case in which we will search for a label. All of the use-cases identified involve iterating over the list and processing all data.

In the example that you suggest - if the user puts 32 labels on the host then at every point they are processed we will need to check all 32 labels. For exporters that are archiving the data - all 32 labels will need to be added to the outgoing stream. For API calls that access the host / charts, all 32 labels will be added as metadata to the outgoing responses.

The only two places that the structure will be touched we will iterate over all the contents:
* At creation time we need to process all the information to create the list.
* At usage time we need to process all of the information to generate the metadata.

An AVL tree will be more complex, and slightly slower, than a linked list for this scenario and so we have already decided to use a linked-list.
